pgdb/v1/schema.go

```diff
@@ -132,6 +142,98 @@ func readIndexes(ctx context.Context, db sqlScanner, desc Descriptor) (map[strin
 	return indexes, nil
 }
 
+// Get a list of the provided descriptor's partition sub tables.
+func ReadPartitionSubTables(ctx context.Context, db sqlScanner, desc Descriptor) ([]string, error) {
+	dialect := goqu.Dialect("postgres")
+
+	qb := dialect.From("pg_inherits")
+	qb = qb.Select("child.relname").As("child")
+	qb = qb.Join(goqu.T("pg_class").As("parent"), goqu.On(goqu.I("pg_inherits.inhparent").Eq(goqu.I("parent.oid"))))
+	qb = qb.Join(goqu.T("pg_class").As("child"), goqu.On(goqu.I("pg_inherits.inhrelid").Eq(goqu.I("child.oid"))))
+	qb = qb.Join(goqu.T("pg_namespace").As("nmsp_parent"), goqu.On(goqu.I("nmsp_parent.oid").Eq(goqu.I("parent.relnamespace"))))
+	qb = qb.Join(goqu.T("pg_namespace").As("nmsp_child"), goqu.On(goqu.I("nmsp_child.oid").Eq(goqu.I("child.relnamespace"))))
+	qb = qb.Where(goqu.L("parent.relname = ?", desc.TableName()))
+	query, params, err := qb.ToSQL()
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := db.Query(ctx, query, params...)
+	if err != nil {
+		return nil, err
+	}
+
+	tables := make([]string, 0)
+	for rows.Next() {
+		var tableName string
+		err = rows.Scan(&tableName)
+		if err != nil {
+			return nil, err
+		}
+		tables = append(tables, tableName)
+	}
+
+	return tables, nil
+}
+
+func tableIsParentPartition(ctx context.Context, db sqlScanner, tableName string) (bool, error) {
+	dialect := goqu.Dialect("postgres")
+
+	qb := dialect.From("pg_class")
+	qb = qb.Select("relname")
+	qb = qb.Join(goqu.T("pg_partitioned_table"), goqu.On(goqu.I("pg_partitioned_table.partrelid").Eq(goqu.I("pg_class.oid"))))
+	qb = qb.Where(goqu.L("relname = ?", tableName))
+	query, params, err := qb.ToSQL()
+	if err != nil {
+		return false, err
+	}
+
+	rows, err := db.Query(ctx, query, params...)
+	if err != nil {
+		return false, err
+	}
+
+	if rows.Next() {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// Get a list of the provided descriptor's partition parent tables.
+func GetParentTables(ctx context.Context, db sqlScanner, desc Descriptor) ([]string, error) {
```

Comment:
is this function needed by the c1 side?  its unused so far?

Reply:
I had a plan for this... I can't recall what is was 😶

Reply:
I removed it for now and if I end up needing it, i'll just re-add it https://github.com/ductone/protoc-gen-pgdb/pull/63/commits/6bcd612aaf415d4c27e94dbae61002b1886e846d